Nama : Yesha Faradina

---
NIM : 240401010072

---
Kelas : IF403





In [47]:
import pandas as pd
import numpy as np

url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df  = pd.read_csv(url)

df.head()


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [48]:
# Q1 : Berapa jumlah baris dan kolom dataset Titanic
print(df.shape, df.columns.tolist())
print(df.isnull().sum())

(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


In [49]:
# Q2 Kolom mana yang memiliki nilai kosong (missing)?
missing_value = df.isnull().sum()
missing_in_precentage = missing_value/len(df)*100

missing_data = pd.DataFrame({
    'Missing' : missing_value,
    'missing precentage' : missing_in_precentage
    })

print(missing_data[missing_data['Missing']> 0 ])



          Missing  missing precentage
Age           177           19.865320
Cabin         687           77.104377
Embarked        2            0.224467


In [50]:
# Q3 Berapa persen penumpang yang selamat (Survived=1)?
survived_passenger_precentage  = df['Survived'].mean()*100
survived_passenger = [df['Survived']==1]


print(f"Penumpang yang selamat adalah", df['Survived'].sum())
print(f"Persentase penumpang yang selamat adalah {survived_passenger_precentage:.3f}%")

Penumpang yang selamat adalah 342
Persentase penumpang yang selamat adalah 38.384%


In [51]:
# Q4 Rata-rata usia penumpang wanita kelas 1 yang selamat?
wk1 = df[
  (df['Sex']=='female') &
  (df['Pclass']==1)
]
print (f"Rata-rata penumpang wanita yang selamat: {wk1['Survived'].mean()*100:.2f}%")
print (f"Rata-rata usia penumpang wanita kleas 1 yang selamat:{wk1['Age'].mean():.2f} tahun")

Rata-rata penumpang wanita yang selamat: 96.81%
Rata-rata usia penumpang wanita kleas 1 yang selamat:34.61 tahun


In [52]:
# Q5 Kelas penumpang mana yang tingkat keselamatannya tinggi?
hasil = df.groupby('Pclass').agg(
    total=('PassengerId', 'count'),
    selamat=('Survived','mean'),
).round(3)
print(hasil)

        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242


In [53]:
# Q6 Tampilkan hasil groupby('Pclass')["Survived"].mean()
hasil = df.groupby('Sex').agg(
    total=('PassengerId', 'count'),
    selamat=('Survived','mean'),
)
print(hasil)

        total   selamat
Sex                    
female    314  0.742038
male      577  0.188908


Berdasarkan data diatas diketahui bahwa dari 891 penumpang, hanya 38,38% yang selamat. Terdapat juga nilai hilang dalam kolom Cabin dan Age yang perlu ditangani. Penumpang kelas 1 memiliki peluang selamat tertinggi (±63%), diikuti kelas 2 (±47%) dan kelas 3 (±24%). Rata-rata usia wanita yang selamat di kelas 1 adalah 34,94 tahun. Secara keseluruhan, status sosial dan demografi memengaruhi tingkat keselamatan.